android/sdl_android/src/main/java/com/smartdevicelink/util/SdlAppInfo.java

```diff
@@ -122,11 +182,93 @@ public String toString() {
         builder.append("\nLast updated: ");
         builder.append(this.lastUpdateTime);
 
+        builder.append("\nVehicle make list: ");
+        builder.append(this.vehicleMakesList.toString());
+
         builder.append("\n-------- Sdl App Info End------");
 
         return builder.toString();
     }
 
+    public static List<VehicleType> deserializeVehicleMake(XmlResourceParser parser) {
+        List<VehicleType> vehicleMakesList = new ArrayList<VehicleType>();
+        try {
+            int eventType = parser.getEventType();
+            while (eventType != XmlPullParser.END_DOCUMENT) {
+                if (eventType == XmlPullParser.START_TAG) {
+                    String tagname = parser.getName();
+                    if (tagname.equalsIgnoreCase("vehicle-type")) {
+                        VehicleType vehicleMake = new VehicleType();
+                        String make = parser.getAttributeValue(null, "make");
+                        if (make != null) {
+                            vehicleMake.setMake(make);
+                            String model = parser.getAttributeValue(null, "model");
+                            if (model != null)
+                                vehicleMake.setModel(model);
+                            String modelYear = parser.getAttributeValue(null, "modelYear");
+                            if (modelYear != null)
+                                vehicleMake.setModelYear(modelYear);
+                            String trim = parser.getAttributeValue(null, "trim");
+                            if (trim != null)
+                                vehicleMake.setTrim(trim);
+                            vehicleMakesList.add(vehicleMake);
+                        }
+                    }
+                }
+                eventType = parser.next();
+            }
+        } catch (XmlPullParserException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return vehicleMakesList;
+    }
+
+    public boolean checkIfVehicleSupported(List<VehicleType> supportedVehicleList, VehicleType connectedVehicle) {
+        if (supportedVehicleList == null || connectedVehicle == null){
+            return false;
```

Comment:
if either of these are false, then the vehicle is supported
```suggestion
            return true;
```

Reply:
@JulianKast could you please point to the things that should be fixed, please? That would help me a lot